rpc/jsonrpc/server/http_server.go

```diff
@@ -280,3 +280,12 @@ func Listen(addr string, maxOpenConnections int) (listener net.Listener, err err
 
 	return listener, nil
 }
+
+// truncateMessage truncate message down to 1K size
+func truncateMessage(message string) string {
```

Comment:
maybe this should take a fmt.Stringer rather than a string. just to be clear. 

my initial fear is that if the thing we put in the message is a JSON string, then we'll truncate in the middle of a JSON document, which would be confusing. 

as a secondary concern, do we want to do this more globally in the logger rather than having to truncate any possibly-long message?

Reply:
Yeah, I feel like this probably should _not_ apply to JSON formatting. Is there some way we could limit this when logs are written to text only?

Reply:
ah, okay. I will check the JSON string format.

for your secondary concern, perhaps we can do it except the `DEBUG` level? 
If this is the direction we would like to go. Then yes!

Reply:
I would just add a `Formatter` in the zerolog and a configuration option for all fields that we log, I think would do this.